src/spartan/upsnark.rs

```diff
@@ -178,18 +184,31 @@ impl<G: Group, EE: EvaluationEngineTrait<G>> RelaxedR1CSSNARKTrait<G> for R1CSSN
     let _ = circuit.synthesize(&mut cs);
 
     // Commits to witness (expensive)
+    let hollow_S = R1CSShape::<G> {
+      num_cons: pk.num_cons_total,
+      num_vars: pk.num_vars_total,
+      num_io: 0,
+      A: vec![],
```

Comment:
Why are A / B / C empty here?

I assume this is because we want to keep the R1CS shape struct untouched for fiatshamir purposes. Either way add a comment describing the reasoning.

Reply:
Added information in the comments. It's just a convenience to keep working with Spartan's existing functions without having to change their signature.